test/suites/route-response-duplicate.spec.ts

```diff
@@ -0,0 +1,134 @@
+      .element(routeResponseLabelInputSelector)
+      .setValue(label);
+    await tests.helpers.duplicateRouteResponse();
+
+    const labelText = await tests.app.client.getValue(
+      routeResponseLabelInputSelector
+    );
+    expect(labelText).to.equal(labelToCompare);
+  });
+
+  it('Verify duplicated route response has the same status code with the original response', async () => {
+    const routeResponseStatusCodeSelectSelector =
+      '.input-group .custom-select[formcontrolname="statusCode"]';
+    const statusCode = '400';
+
+    await tests.helpers.selectRouteResponse(1);
+    await tests.helpers.removeRouteResponse();
+    await tests.app.client
+      .element(routeResponseStatusCodeSelectSelector)
+      .selectByValue(statusCode);
+    await tests.helpers.duplicateRouteResponse();
+
+    const statusCodeValue = await tests.app.client.getValue(
+      routeResponseStatusCodeSelectSelector
+    );
+    expect(statusCodeValue).to.equal(statusCode);
+  });
+
+  it('Verify duplicated route response has the same body with the original response', async () => {
+    const routeResponseAceContentSelector = '.ace_content';
+
+    await tests.helpers.selectRouteResponse(1);
+    await tests.helpers.removeRouteResponse();
+    const originalBodyText = await tests.app.client.getText(
+      routeResponseAceContentSelector
+    );
+    await tests.helpers.duplicateRouteResponse();
+
+    const duplicatedBodyText = await tests.app.client.getText(
+      routeResponseAceContentSelector
+    );
+    expect(duplicatedBodyText).to.equal(originalBodyText);
+  });
+
+  it('Verify duplicated route response has the same headers with the original response', async () => {
+    await tests.helpers.selectRouteResponse(1);
+    await tests.helpers.removeRouteResponse();
+    await tests.helpers.switchTab('HEADERS');
+    await tests.helpers.addHeader(
+      'route-response-headers',
+
+      { key: 'test', value: 'test' }
+    );
+    await tests.helpers.duplicateRouteResponse();
+    await tests.helpers.switchTab('HEADERS');
+
+    await tests.app.client
+      .elements('#route-response-headers .row.headers-list')
+      .should.eventually.have.property('value')
+      .to.be.an('Array')
+      .that.have.lengthOf(2);
+
+    const selectorAndValueAssertionPairs = {
+      '#route-response-headers .row.headers-list:first-child .form-control:first-child':
+        'Content-Type',
+      '#route-response-headers .row.headers-list:first-child .form-control:nth-child(2)':
+        'application/json',
+      '#route-response-headers .row.headers-list:nth-child(2) .form-control:first-child':
+        'test',
+      '#route-response-headers .row.headers-list:nth-child(2) .form-control:nth-child(2)':
+        'test'
+    };
+
+    Object.keys(selectorAndValueAssertionPairs).forEach((selector: string) => {
+      const valueToCompare = selectorAndValueAssertionPairs[selector];
+      tests.helpers.assertElementValue(selector, valueToCompare);
+    });
```

Comment:
This code cannot work. If you test with any wrong value you will see that the test is still passing. There is no await on the helper on line 99. But you cannot just put an async on the `forEach` callback. What you need to do is to put a complete `it('', async () =>{])` inside the forEach. 
You can also see a warning saying "Unhandled promise rejection warning" which means that the test failed, but the failure wasn't caught (because the test did not wait). 

It's also fine for me if you get rid of the `forEach` and just call the helper four times. As long as it's clean and readable it does not need to be that DRY.

Reply:
Forgetting that `await` is bad on my side :) Instead of calling it four times or declaring `it` blocks with each `forEach` I've used `for...of` here. Did the work. Is that okay?

Reply:
I suppose it's better like you did as the assertions are all part of the same test.